baybe/exceptions.py

```diff
@@ -67,3 +67,7 @@ class ModelNotTrainedError(Exception):
 
 class UnmatchedAttributeError(Exception):
     """An attribute cannot be matched against a certain callable signature."""
+
+
+class InvalidSurrogateModelError(Exception):
```

Comment:
IMO, the description and the name together cause confusion. From the name, I would expect this exception to happen when I provide a mis-configured model or something, which is different from a model that behaves different as required, since the model itself can still be configured correctly. Could you specify what you mean here?

Reply:
Your meaning of invalid is in terms of wrong configuration, my meaning is in terms of wrong behavior. If you look at where it is used, it currently covers two cases:
* We request a batch recommendation from a surrogate model that can only offer single recommendations
* A surrogate model provides its predictions in a format different than required (is relevant for the ensemble surrogates).

But I'm happy to rename if we can find something more precise. Can you suggest an alternative?

Reply:
How about changing the docstring to "An invalid surrogate model was chosen"? That fits the name of the exception and also covers the two cases imo

Reply:
changed 👍🏼